packages/react-router/lib/hooks.tsx

```diff
@@ -1046,14 +1046,14 @@ export function useNavigation() {
 export function useRevalidator() {
   let dataRouterContext = useDataRouterContext(DataRouterHook.UseRevalidator);
   let state = useDataRouterState(DataRouterStateHook.UseRevalidator);
+  let revalidate = React.useCallback(
+    async () => dataRouterContext.router.revalidate(),
+    [dataRouterContext.router]
```

Comment:
I'm pretty sure this is no more stable than before

- Before `dataRouterContext.router` was a dependency to the `React.useMemo` below.
- Now `dataRouterContext.router` is a dependency to the `React.useCallback` for `revalidate
- Now `revalidate` is a dependency to the `React.useMemo` below
- Effectively `dataRouterContext.router` is still a dependency of the `React.useMemo` below

You could narrow and make `dataRouterContext.router.revalidate` the dependency. I'm not sure how stable it is

Reply:
Previously, when you kicked off a revalidation and `state.revalidation` went into a `loading` state, you got a new instance of the `revalidate` function

Reply:
Ahh yep, I see it now, thanks!